core/lib/multivm/src/versions/vm_latest/tests/l1_l2_complete_tx_execution.rs

```diff
@@ -0,0 +1,500 @@
+use ethabi::{encode, Contract, Token};
+use zksync_basic_types::{web3::keccak256, Address, H256};
+use zksync_system_constants::{
+    L2_ASSET_ROUTER_ADDRESS, SETTLEMENT_LAYER_RELAY_SENDER, SHARED_BRIDGE_ETHER_TOKEN_ADDRESS,
+};
+use zksync_test_account::Account;
+use zksync_types::{
+    diamond::{compile_initial_cut_hash, facet_cut, Action, ChainCreationParams, VerifierParams},
+    mailbox::BridgeHubRequestL2TransactionOnGateway,
+    U256,
+};
+use zksync_utils::u256_to_h256;
+
+use crate::{
+    interface::TxExecutionMode,
+    vm_latest::{
+        tests::{
+            tester::{VmTester, VmTesterBuilder},
+            utils::{
+                deploy_and_verify_contract, read_admin_facet, read_bridgehub, read_diamond,
+                read_diamond_init, read_diamond_proxy, read_mailbox_facet, read_message_root,
+                read_stm, read_transparent_proxy, send_l2_tx_and_verify, send_prank_tx_and_verify,
+                undo_l1_to_l2_alias, BASE_SYSTEM_CONTRACTS,
+            },
+        },
+        HistoryEnabled,
+    },
+};
+
+fn prepare_environment_and_deploy_contracts(
+    vm: &mut VmTester<HistoryEnabled>,
+    deploy_account_address: Address,
+) -> (Address, Contract) {
+    let mut deploy_nonce = 0;
+
+    // Deploy Bridgehub
+
+    let (bridgehub_contract_bytecode, _bridgehub_contract) = read_bridgehub();
+    let l1_chain_id = Token::Uint(0.into()); // Chain ID hasn't been set
```

Comment:
probably better to say that the chain id does not matter for the purpose of the test